services/sysinfo/client/client.go

```diff
@@ -103,3 +105,90 @@ func (p *uptimeCmd) Execute(ctx context.Context, f *flag.FlagSet, args ...interf
 	}
 	return retCode
 }
+
+type dmesgCmd struct {
+	tail        int64
+	grep        string
+	ignoreCase  bool
+	invertMatch bool
+}
+
+func (*dmesgCmd) Name() string     { return "dmesg" }
+func (*dmesgCmd) Synopsis() string { return "View the messages in kernel ring buffer" }
+func (*dmesgCmd) Usage() string {
+	return `demsg [--tail=N] [--grep=PATTERN] [-i] [-v]:
+	 Print the messages from kernel ring buffer.
+`
+}
+
+func (p *dmesgCmd) SetFlags(f *flag.FlagSet) {
+	f.StringVar(&p.grep, "grep", "", "regual expression filter")
+	f.Int64Var(&p.tail, "tail", -1, "tail the latest n lines")
+	f.BoolVar(&p.ignoreCase, "i", false, "ignore case")
+	f.BoolVar(&p.invertMatch, "v", false, "invert match")
+}
+
+func (p *dmesgCmd) Execute(ctx context.Context, f *flag.FlagSet, args ...interface{}) subcommands.ExitStatus {
+	state := args[0].(*util.ExecuteState)
+	c := pb.NewSysInfoClientProxy(state.Conn)
+
+	req := &pb.DmesgRequest{
+		TailLines:   int32(p.tail),
+		Grep:        p.grep,
+		IgnoreCase:  p.ignoreCase,
+		InvertMatch: p.invertMatch,
+	}
+	stream, err := c.DmesgOneMany(ctx, req)
+	if err != nil {
+		// Emit this to every error file as it's not specific to a given target.
+		for _, e := range state.Err {
+			fmt.Fprintf(e, "All targets - could not info servers: %v\n", err)
+		}
+		return subcommands.ExitFailure
+	}
+
+	// printed timestamp will be the following format: date time timezone
+	// date: YYYY-MM-DD
+	// time: HH-MM-SS
+	// timezone: MST/PDT etc.
+	timestampFormat := "2006-01-02 15:04:05 MST"
```

Comment:
Should we use `time.RFC3339` or another one of the predefined outputs instead of inventing our own?

Reply:
Then I think I will use the default format of time.Time which are also used in file stat action